cloud_Azure/terraform/module/storage_account.tf

```diff
@@ -1,15 +1,25 @@
+# Prepare names that meet Azure Storage Account naming restrictions (only alphanum letters, max 24 length, Azure-wide unique)
+# Each output name is concatenation of Resource Group name and Subscription ID, adjusted to naming restrictions
+locals {
+  _names                          = [for name in var.resource_group_names : "${name}${var.subscription_id}"]
+  _lowercase_names                = [for name in local._names : lower(name)]
+  _alphanum_lowercase_names       = [for name in local._lowercase_names : join("", regexall("[[:alnum:]]+", name))]
+  generated_storage_account_names = [for name in local._alphanum_lowercase_names : substr(name, 0, 24)]
```

Comment:
Are those names guaranteed to be unique after the truncation to 24 characters?

Reply:
there is a tradeoff here - names are more readable when they contain resource group name, but also more likely to duplicate
the remedy is to provide custom storage account names
suggestions on how to improve uniqueness are welcome